google/auth/transport/requests.py

```diff
@@ -246,7 +254,7 @@ def request(
         data=None,
         headers=None,
         max_allowed_time=None,
-        timeout=None,
+        timeout=_DEFAULT_TIMEOUT,
```

Comment:
This can still be overridden by `None` if `None` is the default in a higher-level library... huh. I don't see any cases when we would actually want requests to wait indefinitely?

Reply:
The docs currently say that the transport-specific default timeout is used. https://google-auth.readthedocs.io/en/latest/reference/google.auth.transport.html?highlight=timeout#google.auth.transport.Request.__call__ Could you go through and update that?

If a higher level library is passing None I think we should fix it in that library.  Someone might have a weird reason to want a `None` timeout... I'm open to other opinions though, since I'm not very familiar with how the handwritten libraries use this.

Reply:
That `Request` class is used for refreshing the `auth` credentials (in the `request()` method) - https://github.com/googleapis/google-auth-library-python/blob/1b9de8dfbe4523f3170e47985ab523cb7865de48/google/auth/transport/requests.py#L236-L240

The "main" request still goes through the `requests.Session` class (the parent class of `AuthorizedSession`).

I agree that a higher-level library should fix that, but I'm not sure if that is feasible in the short term (talking for `storage` where the issue has been reported). I err on the side of fixing the user bug sooner rather than later, with the assumption that a desire to have a `None` timeout is a very marginal use case.

I might be wrong, of course, thus any insight of the higher-level library maintainers would be much appreciated.

Reply:
Whoops, that was a mistake on my part. I would still like input from @crwilcox and perhaps others on where this change should be made. If it will be a faster fix here than we can go forward with the default timeout